src/datasets/arrow_dataset.py

```diff
@@ -1994,6 +1994,11 @@ def remove_columns(self, column_names: Union[str, List[str]], new_fingerprint: O
             features: ['text'],
             num_rows: 1066
         })
+        >>> ds.remove_columns(column_names=ds.column_names) # Removing all the columns returns an empty dataset, and the `num_rows` information is lost
```

Comment:
"is lost" is maybe a bit too harsh

```suggestion
        >>> ds.remove_columns(column_names=ds.column_names) # Removing all the columns returns an empty dataset with the `num_rows` property set to 0
```